### PR TITLE
Build release tools binary and slave image

### PR DESCRIPTION
**Pull request by csrwng**
Adds template and pipeline to build the go release-ci tool

### CODE REVIEW

**Comment by csrwng** [comment]
@stevekuznetsov @kargakis ptal

**Comment by csrwng** [comment]
@stevekuznetsov using the release-ci:binaries output from this build, we can use at as image source to add to other images as necessary

**Comment by stevekuznetsov** [comment]
Cool, this looks good.

File: tools/build/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+```
+oc new-app -f https://raw.githubusercontent.com/openshift/release/master/tools/build/build-tools.yaml
+```
+
+Optional template parameters are:
+- RELEASE_URL - git URL of the source repository to build from
+- RELEASE_REF - git ref of the source repository to build from
+
+For example, to build from one your branches:
```
Comment (kargakis) [inline tools/build/README.md]: typo

**Comment by kargakis** [comment]
It seems that we will need to have a controller that knows how to trigger JenkinsPipeline BCs as opposed to querying Jenkins directly.

**Comment by csrwng** [comment]
@kargakis I don't follow ... we already do that (trigger Jenkins pipelines automatically) with the BC config trigger

**Comment by kargakis** [comment]
@csrwng yeah, I was commenting on having jobs in prow that run jenkins pipelines. Today the flow is:

1.  trusted PR opens OR org member comments /ok-to-test OR org member comments /test ... or /retest OR a commit is pushed in a branch
2. prowjob is created via github event
3. plank notices new prowjob and start jenkins job

Assuming we want to support triggering jenkins pipelines in any of the aforementioned cases (1.), the flow needs to be:
1.  trusted PR opens OR org member comments /ok-to-test OR org member comments /test ... or /retest OR a commit is pushed in a branch
2. prowjob is created via github event
3. new controller notices new prowjob and starts a build for JenkinsPipeline BC
4. BC controller is responsible for interacting with Jenkins?

**Comment by csrwng** [comment]
Ahh I see what you're saying... but I don't think we gain much by triggering the job through the BC instead of kicking it off in Jenkins directly.

**Comment by kargakis** [comment]
Maybe. I haven't looked closely. If we can reuse the existing controller, that would be great.

File: tools/build/README.md (hunk 1 above)
Comment (csrwng) [inline tools/build/README.md]: grammar fixed

**Comment by stevekuznetsov** [comment]
Yeah, it looks like we should be fine to just trigger the Jenkins pipeline from the existing controller.

**Pull request merged**